### PR TITLE
Deprecating statement refinement syntax - #2765

### LINKED ISSUE #2765: Deprecate statement level refinements
This is extracting a portion of Issue #2425 to just the matter of deprecating refinement of statements, such as if, while, assert, assume.

### PR DESCRIPTION
**Pull request by davidcok**
Fixes #2765, which was part of #2425

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: Makefile
Comment (MikaelMayer) [inline Makefile:38]: Why do you need the -r and the -f flag? -r is for recursive, and -f is for force, and I don't think you need either of these.

**Review by MikaelMayer — COMMENTED** [review]
I'm good with it, but I would prefer to have @RustanLeino provide the final review for that because I don't know the ins and outs of statement refinements

File: Makefile
Comment (davidcok) [reply Makefile:38]: The -r is because obj is a directory; the -f is because the arguments might not exist.
The reason for this addition is that the location of the Scanner.cs and Parser.cs files changed at some point. If one switches branches, does a `make clean exe`, errors or an incorrect build will result because of the obsolete Scanner or Parser. I've been caught be this many times and I often have forgotten the problem

**Comment by davidcok** [comment]
OK with having Rustan review it, Actually Anjali has called a halt on deprecating stuff, so there will be some strategic conversation before this is merged.

**Review by RustanLeino — CHANGES_REQUESTED** [review]
Looks good. I suggest that this PR also deprecate the `modify` statement.

**Comment by davidcok** [comment]
Did you mean to deprecate both senses of the modify statement?

**Comment by davidcok** [comment]
Mikael - is it the `modify X` (i.e. havoc) that you use, or the `modify x { }`  which ought to be `modifies` and which Rustan proposes to perhaps eventually become do requires... modifies... ensures... { ... }?

**Comment by RustanLeino** [comment]
Okay, let's treat the `modify` statement separately from this PR. I will approve this PR as is.

**Review by RustanLeino — APPROVED** [review]

**Pull request merged**
